python/pyarrow/tests/parquet/test_dataset.py

```diff
@@ -938,46 +671,26 @@ def test_read_multiple_files(tempdir, use_legacy_dataset):
     (dirpath / '_SUCCESS.crc').touch()
 
     def read_multiple_files(paths, columns=None, use_threads=True, **kwargs):
-        dataset = pq.ParquetDataset(
-            paths, use_legacy_dataset=use_legacy_dataset, **kwargs)
+        dataset = pq.ParquetDataset(paths, **kwargs)
         return dataset.read(columns=columns, use_threads=use_threads)
 
     result = read_multiple_files(paths)
     expected = pa.concat_tables(test_data)
 
     assert result.equals(expected)
 
-    # Read with provided metadata
-    # TODO(dataset) specifying metadata not yet supported
```

Comment:
We actually do support that through a different dataset API (`pyarrow.dataset.parquet_dataset`), but that is not wired up in the parquet API. Given that we never got any complaints about this, I suppose fine to just remove (and it should already be tested in the dataset tests, so should be fine to remove the tests here as well)